src/malcolm/reducer/method.reducer.test.js

```diff
@@ -36,4 +36,212 @@ describe('method reducer', () => {
     );
     expect(attribute.inputs.input1).toEqual(123);
   });
+
+  it('updateMethodInput should flag an input on a method as dirty', () => {
+    const state = {
+      blocks: {
+        block1: {
+          attributes: [
+            {
+              name: 'attr1',
+              inputs: {},
+            },
+          ],
+        },
+      },
+    };
+
+    const payload = {
+      path: ['block1', 'attr1'],
+      name: 'input1',
+      value: { isDirty: true },
+    };
+
+    const action = {
+      type: MalcolmUpdateMethodInputType,
+      payload,
+    };
+
+    const updatedState = MethodReducer(state, action);
+    const attribute = blockUtils.findAttribute(
+      updatedState.blocks,
+      'block1',
+      'attr1'
+    );
+    expect(attribute.dirtyInputs.input1).toBeTruthy();
+  });
+
+  it('updateMethodInput should update dirty state of an input on a method', () => {
+    const state = {
+      blocks: {
+        block1: {
+          attributes: [
+            {
+              name: 'attr1',
+              inputs: {},
+              dirtyInputs: {
+                input1: false,
+              },
+            },
+          ],
+        },
+      },
+    };
+
+    const payload = {
+      path: ['block1', 'attr1'],
+      name: 'input1',
+      value: { isDirty: true },
+    };
+
+    const action = {
+      type: MalcolmUpdateMethodInputType,
+      payload,
+    };
+
+    const updatedState = MethodReducer(state, action);
+    const attribute = blockUtils.findAttribute(
+      updatedState.blocks,
+      'block1',
+      'attr1'
+    );
+    expect(attribute.dirtyInputs.input1).toBeTruthy();
+  });
+
+  it('handleMethodReturn should update the output on a method with return map', () => {
+    const state = {
```

Comment:
you could set a basic state in `beforeEach` and then update the bits that change